pkg/deployment/bootstrap.go

```diff
@@ -0,0 +1,129 @@
+//
+// DISCLAIMER
+//
+// Copyright 2018 ArangoDB GmbH, Cologne, Germany
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+// Author Ewout Prangsma
+//
+
+package deployment
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+
+	api "github.com/arangodb/kube-arangodb/pkg/apis/deployment/v1alpha"
+	"github.com/arangodb/kube-arangodb/pkg/util/k8sutil"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	driver "github.com/arangodb/go-driver"
+
+	"github.com/arangodb/kube-arangodb/pkg/util/constants"
+)
+
+const (
+	rootUserName = "root"
+)
+
+// EnsureBootstrap executes the bootstrap once as soon as the deployment becomes ready
+func (d *Deployment) EnsureBootstrap() error {
+
+	status, version := d.GetStatus()
+
+	if status.Conditions.IsTrue(api.ConditionTypeReady) {
+		if status.Conditions.IsTrue(api.ConditionTypeBoostrapCompleted) {
+			return nil // Nothing to do, already bootstrapped
+		}
+
+		d.deps.Log.Info().Msgf("Bootstrap deployment %s", d.Name())
```

Comment:
This will likely run once for existing deployments.

That sounds like a problem.